src/commandPalette/createAlert.ts

```diff
@@ -0,0 +1,145 @@
+  const extensionFile = getExtensionFilePath(context)!;
+  const extension: ExtensionStub = yaml.parse(
+    readFileSync(extensionFile).toString()
+  );
+
+  if (!extension.metrics) {
+    vscode.window.showWarningMessage(
+      "No metrics defined in extension.yaml, please define them before creating alerts"
+    );
+    return;
+  }
+
+  // TODO, we could ask the user if they want to create a new alert or edit an existing one?
+
+  // Ask the user to select metric
+  const metricKeys = extension.metrics.map((metric) => metric.key);
+  const metricToUse = await vscode.window.showQuickPick(metricKeys, {
+    placeHolder: "Choose a metric",
+    title: "Extension workspace: Create Alert",
+  });
+  if (!metricToUse) {
+    vscode.window.showErrorMessage(
+      "No metric was selected. Operation cancelled."
+    );
+    return;
+  }
+
+  // Ask the user to input the alert name
+  const alertName = await vscode.window.showInputBox({
+    placeHolder: `Alert name for ${metricToUse}`,
+    title: "Extension workspace: Create Alert",
+  });
+  if (!alertName) {
+    vscode.window.showErrorMessage(
+      "No alert name was entered. Operation cancelled."
+    );
+    return;
+  }
+
+  // Ask the user if the condition is ABOVE or BELOW
+  const alertCondition = await vscode.window.showQuickPick(["ABOVE", "BELOW"], {
+    placeHolder: `Alert condition for ${metricToUse}`,
+    title: "Extension workspace: Create Alert",
+  });
+  if (!alertCondition) {
+    vscode.window.showErrorMessage(
+      "No alert condition was selected. Operation cancelled."
+    );
+    return;
+  }
+
+  // Ask the user to input the threshold
+  const threshold = await vscode.window.showInputBox({
+    placeHolder: `Threshold for ${metricToUse} when ${alertCondition}`,
+    title: "Extension workspace: Create Alert",
+  });
+
+  if (!threshold || isNaN(Number(threshold))) {
+    vscode.window.showErrorMessage(
+      "No valid threshold was entered. Operation cancelled."
+    );
+    return;
+  }
+
+  // Convert threshold to a number
+  const numberThreshold = Number(threshold);
+
+
+  // Create directories for alerts if they don't exist
+  const extensionDir = path.resolve(extensionFile, "..");
+  const alertsDir = path.resolve(extensionDir, "alerts");
+  if (!existsSync(alertsDir)) {
+    mkdirSync(alertsDir);
+  }
+
+  // Count how many files we have inside the alerts directory with readDirSync
+  // The current file number is that number + 1
+  const currentFileNumber = readdirSync(alertsDir).length + 1;
+  // Pad the number with zeros so the lenght is always 3
+  const paddedFileNumber = currentFileNumber.toString().padStart(3, "0");
+
+  // Convert alertName to lowerCase, only allow \w and - characters
+  // Make sure we don't have multiple - in a row
+  const alertNameForFile = alertName.toLowerCase().replace(/[^a-z0-9-]/g, "-").replace(/-+/g, "-");
+  
+  const fileName = `alert-${paddedFileNumber}-${alertNameForFile}.json`;
```

Comment:
Not foolproof: if I  have alerts  `xxx-001`, `xxx-002`, and `xxx-003` generated, then I delete alert `xxx-002`  and I run the flow to generate a new one I will overwrite my `xxx-003` alert. 

Could we maybe do one of:
* get a short hash
* grab the latest "number" part of any alert file name and just keep increasing by 1
* add check if name exists and prompt user for overwrite vs. rename?

Reply:
I've pushed a fix to always generate unique names. If the name already exists the counter is increased

I don't agree with hard coding a check that 10 alerts have been reached, we should let the API (or the schema definition) do that when uploading the extension. What happens if this number changes?

Reply:
Fair enough. I thought we could have done an earlier check and avoid a failed build.
But you're right, there'd be no way for us to dynamically grab the limit.